test/lib/zeitwerk/test_push_dir.rb

```diff
@@ -19,4 +19,14 @@ class TesPushDir < LoaderTest
     e = assert_raises(ArgumentError) { loader.push_dir(dir) }
     assert_equal "the root directory #{dir} does not exist", e.message
   end
+
+  module MyApp
+  end
+
+  test "an alternative namespace can be passed" do
+    files = [["app/x.rb", "TesPushDir::MyApp::X = true"]]
+    with_setup(files, dirs: [['app', namespace: MyApp]]) do
```

Comment:
One likely footgun here is that if `MyApp` is reloadable, we'd hold onto a dereferenced constant.